src/amoc_scenario.erl

```diff
@@ -80,59 +90,67 @@ handle_cast(_, State) ->
 %%-------------------------------------------------------------------------
 -spec start_scenarios_ets() -> amoc_scenarios.
 start_scenarios_ets() ->
-    ets:new(amoc_scenarios, [named_table,
-                             protected,
-                             {read_concurrency, true}]).
+    EtsOptions = [named_table, protected, {read_concurrency, true}],
+    ets:new(amoc_scenarios, EtsOptions),
+    ets:new(uploaded_modules, EtsOptions).
 
 -spec add_code_paths() -> ok | {error, {bad_directories, [file:filename()]}}.
 add_code_paths() ->
     true = code:add_pathz(?EBIN_DIR),
-    AdditionalCodePaths = amoc_config_env:get(extra_code_paths, []),
+    AdditionalCodePaths = amoc_config_env:get(amoc, extra_code_paths, []),
     Res = [{code:add_pathz(Path), Path} || Path <- [?EBIN_DIR | AdditionalCodePaths]],
     case [Dir || {{error, bad_directory}, Dir} <- Res] of
         [] -> ok;
         BadDirectories -> {error, {bad_directories, BadDirectories}}
     end.
 
--spec find_scenario_modules() -> [module()].
+-spec find_scenario_modules() -> any().
 find_scenario_modules() ->
     AllBeamFiles = [File || Path <- code:get_path(), File <- filelib:wildcard("*.beam", Path)],
     AllModules = [list_to_atom(filename:rootname(BeamFile)) || BeamFile <- AllBeamFiles],
     ok = code:ensure_modules_loaded(AllModules),
-    AllScenarios = [Module || Module <- erlang:loaded(), is_scenario(Module)],
-    [ets:insert(amoc_scenarios, {Scenario, scenario, preloaded}) || Scenario <- AllScenarios].
+    [maybe_store_module(Module) || Module <- erlang:loaded()].
+
+-spec maybe_store_module(module()) -> any().
+maybe_store_module(Module) ->
+    case get_module_type(Module) of
+        scenario ->
+            ets:insert(amoc_scenarios, {Module, scenario});
+        parametrised ->
+            ets:insert(amoc_scenarios, {Module, parametrised});
+        ordinary ->
+            ok
+    end.
 
--spec is_scenario(module()) -> boolean().
-is_scenario(Module) ->
+-spec get_module_type(module()) -> scenario | parametrised | ordinary.
+get_module_type(Module) ->
     case erlang:function_exported(Module, module_info, 1) of
-        false -> false;
+        false -> ordinary;
```

Comment:
Is there a module that doesn't export the `module_info` function?

Reply:
it's surprising but yes :)

Reply:
All right, thanks for the explanation. Could you put this in a comment in the code?